src/PlayFabBuddy.Infrastructure/Repositories/LocalMasterPlayerAccountRepository.cs

```diff
@@ -50,30 +51,49 @@ public List<MasterPlayerAccountEntity> Get()
             throw new NullReferenceException("Entity List, is null, Error on Deserialization");
         }
 
+        var aggregates = new List<MasterPlayerAccountAggregate>();
+
+        foreach (var entity in entityList)
+        {
+            aggregates.Add(new MasterPlayerAccountAggregate(entity));
+        }
+
         _cache.Clear();
-        _cache.AddRange(entityList);
+        _cache.AddRange(aggregates);
 
         return _cache;
 
     }
 
-    public async Task Save(List<MasterPlayerAccountEntity> toSave)
+    public async Task Save(List<MasterPlayerAccountAggregate> toSave)
     {
+        var entityList = ConvertToEntity(toSave);
         await using var writeStream = File.Create(_configPath);
-        await JsonSerializer.SerializeAsync(writeStream, toSave, _jsonOptions);
+        await JsonSerializer.SerializeAsync(writeStream, entityList, _jsonOptions);
         _cache.Clear();
         _cache.AddRange(toSave);
         await writeStream.DisposeAsync();
     }
 
-    public async Task Append(List<MasterPlayerAccountEntity> toAppend)
+    public async Task Append(List<MasterPlayerAccountAggregate> toAppend)
     {
+        var entityList = ConvertToEntity(toAppend);
         var oldUsers = Get();
         oldUsers.AddRange(toAppend);
         await using var writeStream = File.Create(_configPath);
-        await JsonSerializer.SerializeAsync(writeStream, oldUsers, _jsonOptions);
+        await JsonSerializer.SerializeAsync(writeStream, entityList, _jsonOptions);
         _cache.Clear();
         _cache.AddRange(oldUsers);
         await writeStream.DisposeAsync();
     }
+
+    private List<MasterPlayerAccountEntity> ConvertToEntity(List<MasterPlayerAccountAggregate> aggregates)
```

Comment:
This should be a method on teh aggregate, rather than on the repo

Reply:
After discussion (offline), this does not fit the Aggregate, so we keep it here for now